src/app/clinic/clinic_facade.cpp

```diff
@@ -10,9 +11,19 @@ void Clinic::appendPatient(const std::shared_ptr<Patient>& patient)
     patients_.emplace_back(patient);
 }
 
-void Clinic::appendVisit(const std::shared_ptr<Visit>& visit)
+void Clinic::appendVisit()
```

Comment:
I suggest name wchich resembles the action conducted by the function: storeTempVisit for example. "appendTempVisit" would even be better, because it actually does that., but the another function of same name needs renaming too. This is even more important, as current appendTempVisit does not append anything - it just rembers the given visit at temp. So it should be named setTempVisit or something alike.